disperser/apiserver/server.go

```diff
@@ -282,7 +283,13 @@ func (s *DispersalServer) disperseBlob(ctx context.Context, blob *core.Blob, aut
 
 	s.logger.Debug("received a new blob dispersal request", "authenticatedAddress", authenticatedAddress, "origin", origin, "blobSizeBytes", blobSize, "securityParams", strings.Join(securityParamsStrings, ", "))
 
-	if s.ratelimiter != nil {
+	// If paymentHeader is not empty, we use the meterer, otherwise we use the ratelimiter if the ratelimiter is available
+	if paymentHeader != nil && *paymentHeader != (core.PaymentMetadata{}) {
```

Comment:
Probably just check for nil and pass in nil for the non-paid endpoint.

Reply:
updated!